glean-core/csharp/Glean/GleanParser.cs

```diff
@@ -0,0 +1,208 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace GleanTasks
+{
+    public class GleanParser : ToolTask
+    {
+        // The name of the directory in which the Python virtual
+        // environment must be created.
+        private const string DefaultVirtualEnvDir = ".venv";
+
+        // The glean_parser pypi package version 
+        private const string GleanParserVersion = "1.28.3";
+
+        // This script runs a given Python module as a "main" module, like
+        // `python -m module`. However, it first checks that the installed
+        // package is at the desired version, and if not, upgrades it using `pip`.
+        //
+        // ** IMPORTANT**
+        // Keep this script in sync with the one in `glean-gradle-plugin/GleanGradlePlugin.groovy`.
+        private const string RunPythonScript = @"
+import importlib
+import subprocess
+import sys
+module_name = sys.argv[1]
+expected_version = sys.argv[2]
+try:
+    module = importlib.import_module(module_name)
+except ImportError:
+    found_version = None
+else:
+    found_version = getattr(module, '__version__')
+if found_version != expected_version:
+    subprocess.check_call([
+        sys.executable,
+        '-m',
+        'pip',
+        'install',
+        '--upgrade',
+        f'{module_name}=={expected_version}'
+    ])
+try:
+    subprocess.check_call([
+        sys.executable,
+        '-m',
+        module_name
+    ] + sys.argv[3:])
+except:
+    # We don't need to show a traceback in this helper script.
+    # Only the output of the subprocess is interesting.
+    sys.exit(1)
+";
```

Comment:
I'm glad we're able to do the same auto-update thing here. That will save some headaches for our users.

Reply:
Yes! I'm a bit unhappy about the duplication, but I couldn't find any easy way to work around this.